scripts/generate_binary_using_dockerfile.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+
+# Find an available container tool (docker or podman)
+find_container_tool() {
+    if command -v docker > /dev/null 2>&1; then
+        echo "sudo docker"
+    elif command -v podman > /dev/null 2>&1; then
+        echo "podman"
+    else
+        echo "Error: Neither docker nor podman is available." >&2
+        exit 1
+    fi
+}
+
+# Determine which container tool to use
+CONTAINER_TOOL=$(find_container_tool)
+
+# Find the directory containing the dockerfile by traversing up the directory tree
+find_dockerfile_dir() {
```

Comment:
I think it would be simpler to get the path to the script using `$(dirname "$0")` and then take `../Dockerfile`. Traversing whole directory tree is not needed.

Reply:
The reason for this is to be able to call the script from where ever you want without being worried to break the `podman build` command.
Thanks to this I can call the script from the download one while being outside the repository dir.

Also, if for some reason you want to move the scripts to other dir, you can be sure that they will not break.

Reply:
If you use `$(dirname "$0")/../Dockerfile`, this will work as long as your keep the script in a subdir at the same level as it is right now.

e.g. If you change the script to `/root/scripts/docker-scripts/generate_binary_using_dockerfile.sh`, the script will break.

But yeah, you are right that it should still work if you call it from any other directory. So, if you are fine with keeping the script always under `/scripts`, I can make the change knowing that it will never break.

Since I didn't know if you will ever move the script to any other dir, I decided to do the find on the directory tree. Now that you are suggesting this, I imagine the directory structure will always be the same, so I'm going to make the change and ping you once done :smile: .